src/libraries/Microsoft.PowerFx.Core/Utils/MutationUtils.cs

```diff
@@ -65,5 +70,44 @@ public static void CheckSemantics(TexlBinding binding, TexlFunction function, Te
                 return;
             }
         }
+
+        public static string GetScalarSingleColumnNameForType(Features features, DKind kind)
```

Comment:
>GetScalarSingleColumnNameForType [](http://example.com/codeflow?start=29&length=32)

Since this is only used in the CollectFunction, can we move it there (as a private method)? There's no need to share it with others if it is not used by more than one function.